platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ProblemTaskIdentityTracker.java

```diff
@@ -25,26 +25,26 @@
  *
  * @see <a href="https://github.com/gradle/gradle/issues/31430">https://github.com/gradle/gradle/issues/31430</a>
  */
-public class ProblemTaskPathTracker {
+public class ProblemTaskIdentityTracker {
 
-    private static final ThreadLocal<String> TASK_NAME = new ThreadLocal<String>();
+    private static final ThreadLocal<TaskIdentity> TASK_IDENTITY = new ThreadLocal<TaskIdentity>();
```

Comment:
https://github.com/gradle/gradle/pull/32328 will add this anyway and it makes the task identity retrieval more explicit.